src/copyeventfilter.cpp

```diff
@@ -0,0 +1,51 @@
+#include "copyeventfilter.h"
+
+#include <QClipboard>
+#include <QGuiApplication>
+#include <QKeyEvent>
+
+CopyEventFilter::CopyEventFilter(QAbstractItemView* view, int role) :
+  CopyEventFilter(view, [=](auto& index) { return index.data(role).toString(); })
+{
+
+}
+
+CopyEventFilter::CopyEventFilter(
+  QAbstractItemView* view, std::function<QString(const QModelIndex&)> format) :
+  QObject(view), m_view(view), m_format(format)
+{
+
+}
+
+bool CopyEventFilter::copySelection() const
+{
+  if (!m_view->selectionModel()->hasSelection()) {
+    return true;
```

Comment:
Should this be `false`?

Reply:
I removed the return value because the only purpose was to stop event propagation in `event`, but it does not make sense here. If there is `Ctrl+C`, event propagation should stop even if the selection is empty (from my point-of-view).